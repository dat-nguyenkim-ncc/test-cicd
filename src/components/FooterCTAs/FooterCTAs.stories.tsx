import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import FooterCTAs, { FooterCTAsProps } from './FooterCTAs'
import { Box } from 'theme-ui'

export default {
  title: 'Search/FooterCTAs',
  component: FooterCTAs,
  parameters: {
    layout: 'padded',
  },
} as Meta

const Template: Story<FooterCTAsProps> = args => (
  <Box sx={{ p: 5, backgroundColor: 'gray03', width: '100%' }}>
    <FooterCTAs {...args} />
  </Box>
)

export const Default = Template.bind({})
Default.args = {
  buttons: [
    {
      label: 'Button 1',
      onClick: () => {},
    },
    {
      label: 'Button 2',
      variant: 'outlineWhite',
      onClick: () => {},
    },
  ],
} as FooterCTAsProps

export const OnlyOneButton = Template.bind({})
OnlyOneButton.args = {
  buttons: [
    {
      label: 'Button 1',
      onClick: () => {},
    },
  ],
} as FooterCTAsProps
