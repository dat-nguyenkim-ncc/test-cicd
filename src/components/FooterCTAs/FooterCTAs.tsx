import React from 'react'
import { Flex } from 'theme-ui'
import { Button } from '../'
import { Variants, ViewInterface } from '../../types'
import { footerZIndex } from '../../utils/consts'

type CTA = {
  label: string
  disabled?: boolean
  onClick?(): void
  variant?: Variants
}

export type FooterCTAsProps = ViewInterface<{
  buttons: CTA[]
}>

const FooterCTAs = ({ buttons, sx }: FooterCTAsProps) => {
  return (
    <Flex
      sx={{
        justifyContent: 'center',
        backgroundColor: 'primary',
        alignItems: 'center',
        borderRadius: 10,
        py: 5,
        mt: 5,
        width: '100%',
        position: 'sticky',
        bottom: 1,
        zIndex: footerZIndex,
        ...sx,
      }}
    >
      {buttons.map((b, index) => (
        <Button
          key={index}
          disabled={b.disabled}
          onPress={b.onClick}
          sx={{ ml: index > 0 ? 4 : 0, width: '290px' }}
          variant={b.variant || 'secondary'}
          label={b.label}
        />
      ))}
    </Flex>
  )
}

export default FooterCTAs
