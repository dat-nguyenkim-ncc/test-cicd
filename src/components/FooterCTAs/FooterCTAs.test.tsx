import * as React from 'react'
import { shallow } from 'enzyme'
import FooterCTAs from './FooterCTAs'

describe('FooterCTAs', () => {
  test('matches snapshot', () => {
    const wrapper = shallow(
      <FooterCTAs
        buttons={[
          {
            label: 'Button 1',
            onClick: () => {},
          },
          {
            label: 'Button 2',
            onClick: () => {},
          },
        ]}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
