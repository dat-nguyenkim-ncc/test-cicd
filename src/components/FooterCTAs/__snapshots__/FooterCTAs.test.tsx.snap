// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterCTAs matches snapshot 1`] = `ShallowWrapper {}`;
